crates/rpc/eth/block.rs

```diff
@@ -193,6 +198,43 @@ impl RpcHandler for GetBlockReceiptsRequest {
     }
 }
 
+impl RpcHandler for GetRawHeaderRequest {
+    fn parse(params: &Option<Vec<Value>>) -> Result<GetRawHeaderRequest, RpcErr> {
+        let params = params.as_ref().ok_or(RpcErr::BadParams)?;
+        if params.len() != 1 {
+            return Err(RpcErr::BadParams);
+        };
+
+        let block: BlockIdentifier = serde_json::from_value(params[0].clone())?;
+        let block_str: String = serde_json::from_value(params[0].clone())?;
+
+        if let BlockIdentifier::Number(_) = block {
+            if !block_str.starts_with("0x") {
+                return Err(RpcErr::BadHexFormat);
+            }
+        }
```

Comment:
Not for this PR, but I think we should add this check to all the others block requests too